pkgs/development/libraries/mkldnn/default.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchFromGitHub, cmake, substituteAll, bash, autoPatchelfHook }:
+
+stdenv.mkDerivation rec {
+  pname = "mkldnn";
+  version = "1.0.2";
+
+  src = fetchFromGitHub {
+    owner = "intel";
+    repo = "mkl-dnn";
+    rev = "v${version}";
+    sha256 = "0lf14a4j511pydyz78wwbl6hifbq86m3dy119iss9cvz8zzrdyvw";
+  };
+
```

Comment:
I would also add different output paths
```suggestion
  outputs = [ "out" "dev" "doc" ];
```

cleans up the outputs quite a bit:
default:
```
$ tree ./results/mkldnn/
./results/mkldnn/
├── include
│   ├── mkldnn_config.h
│   ├── mkldnn_debug.h
│   ├── mkldnn.h
│   ├── mkldnn.hpp
│   ├── mkldnn_types.h
│   └── mkldnn_version.h
├── lib
│   ├── cmake
│   │   └── mkldnn
│   │       ├── mkldnn-config.cmake
│   │       ├── mkldnn-config-version.cmake
│   │       ├── mkldnn-targets.cmake
│   │       └── mkldnn-targets-release.cmake
│   ├── libmkldnn.so -> libmkldnn.so.1
│   ├── libmkldnn.so.1 -> libmkldnn.so.1.0
│   └── libmkldnn.so.1.0
└── share
    └── doc
        └── mkldnn
            └── LICENSE

7 directories, 14 files
```


Post multi-output "out":
```
$ tree ./result
./result
└── lib
    ├── libmkldnn.so -> libmkldnn.so.1
    ├── libmkldnn.so.1 -> libmkldnn.so.1.0
    └── libmkldnn.so.1.0
```
post-multi-output dev:
```
$ tree ./result-dev
./result-dev
├── include
│   ├── mkldnn_config.h
│   ├── mkldnn_debug.h
│   ├── mkldnn.h
│   ├── mkldnn.hpp
│   ├── mkldnn_types.h
│   └── mkldnn_version.h
├── lib
│   └── cmake
│       └── mkldnn
│           ├── mkldnn-config.cmake
│           ├── mkldnn-config-version.cmake
│           ├── mkldnn-targets.cmake
│           └── mkldnn-targets-release.cmake
└── nix-support
    └── propagated-build-inputs
```
post-multi-output doc:
```
$ tree ./result-doc
./result-doc
└── share
    └── doc
        └── mkldnn
            └── LICENSE

3 directories, 1 file
```

Also saves a small amount of space for the default closure:
```
[nix-shell:~/.cache/nix-review/pr-68014]$ nix path-info -Sh ./results/mkldnn
/nix/store/zrrj671vf2l04p79r8sc39idjy3fbhd0-mkldnn-1.0.2          49.2M

[nix-shell:~/.cache/nix-review/pr-68014]$ nix path-info -Sh ./result
/nix/store/r22kq0jvvxgndcyywvddgcm7sfc2cq4n-mkldnn-1.0.2          48.8M
```